File: formula-js/src/FormulaRandom.test.ts
import {FormulaRandom} from "./FormulaRandom";

test('random int', () => {
  const seed = 10;
  const random = new FormulaRandom(seed);

  expect(random.nextInt()).toBe(297746560);
  expect(random.nextInt()).toBe(1120512893);
});

test('random int (different seed)', () => {
  const seed = 12345;
  const random = new FormulaRandom(seed);

  expect(random.nextInt()).toBe(1406938949);
  expect(random.nextInt()).toBe(506849219);
});

test('random float', () => {
  const seed = 10;
  const random = new FormulaRandom(seed);

  expect(random.nextFloat()).toBeCloseTo(0.13864);
  expect(random.nextFloat()).toBeCloseTo(0.52177);
});

test('random range', () => {
  const seed = 12345;
  const random = new FormulaRandom(seed);

  expect(random.nextRange(5, 10)).toBe(8);
  expect(random.nextRange(5, 10)).toBe(6);
});

File: java/formula-java/src/test/java/org/formula/FormulaRandomTest.java
package org.formula;

import static org.assertj.core.api.Assertions.assertThat;

import org.assertj.core.data.Offset;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;

class FormulaRandomTest {

    @Test
    @DisplayName("random int")
    void randomInt() {
        long seed = 10L;
        var random = new FormulaRandom(seed);

        assertThat(random.nextInt()).isEqualTo(297746560L);
        assertThat(random.nextInt()).isEqualTo(1120512893L);
    }

    @Test
    @DisplayName("random int (different seed)")
    void randomIntDifferentSeed() {
        long seed = 12345L;
        var random = new FormulaRandom(seed);

        assertThat(random.nextInt()).isEqualTo(1406938949L);
        assertThat(random.nextInt()).isEqualTo(506849219L);
    }

    @Test
    @DisplayName("random float")
    void randomFloat() {
        long seed = 10L;
        var random = new FormulaRandom(seed);

        assertThat(random.nextFloat()).isCloseTo(0.13864f, Offset.offset(0.005f));
        assertThat(random.nextFloat()).isCloseTo(0.52177f, Offset.offset(0.005f));
    }
}